ccvm/workload.go

```
@@ -136,19 +158,158 @@ func loadWorkloadData(ctx context.Context, ws *workspace, workloadName string) (
+		wkld.spec.BaseImageName = parent.spec.BaseImageName
+	}
+
+	if wkld.spec.Hostname == "" {
+		wkld.spec.Hostname = parent.spec.Hostname
+	}
+
+	// Always better to require nested VM that not.
+	if !wkld.spec.NeedsNestedVM {
+		wkld.spec.NeedsNestedVM = parent.spec.NeedsNestedVM
+	}
+
+	wkld.spec.VM.Merge(&parent.spec.VM)
+}
+
+type cloudConfig map[string]interface{}
+
+func (cc cloudConfig) merge(p cloudConfig) error {
+	for k, v := range p {
+		// copy from parents to child if not present
+		if _, ok := cc[k]; !ok {
+			cc[k] = v
+			continue
+		}
+
+		// else merge slices and maps
+		switch reflect.ValueOf(v).Type().Kind() {
+		case reflect.Slice:
+			if reflect.ValueOf(cc[k]).Type() != reflect.ValueOf(v).Type() {
+				return fmt.Errorf("Types of merged sequences not equal for: %s", k)
+			}
+
+			// create new slice appending current slice (cc[k]) onto the parent slice (v)
+			s := reflect.AppendSlice(reflect.ValueOf(v), reflect.ValueOf(cc[k]))
+
+			// update the top-level map (cc) for the current key (k) with the temporary slice (s)
+			reflect.ValueOf(cc).SetMapIndex(reflect.ValueOf(k), s)
+		case reflect.Map:
+			// first level merging only
+			if reflect.ValueOf(cc[k]).Type() != reflect.ValueOf(v).Type() {
+				return fmt.Errorf("Types of merged maps not equal for: %s", k)
+			}
+
+			// for each key (kk) in the parent map (v)...
+			for _, kk := range reflect.ValueOf(v).MapKeys() {
+				// ... check if the key (kk) is present in the current map (cc[k])
+				if !reflect.ValueOf(cc[k]).MapIndex(kk).IsValid() {
+					// ... and if not copy into the current map (cc[k])
+					reflect.ValueOf(cc[k]).SetMapIndex(kk, reflect.ValueOf(v).MapIndex(kk))
+				}
+			}
+		}
+
+		// and for everything else just use version in cc
+	}
+	return nil
+}
+
+func (wkld *workload) parse(ws *workspace) (cloudConfig, error) {
+	var p cloudConfig
+	var err error
+	if wkld.parent != nil {
+		p, err = wkld.parent.parse(ws)
+		if err != nil {
+			return nil, nil
```

Why do we ignore the error here?

---

Dunno :-) Fixed.